spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinderTests.java

```diff
@@ -71,4 +94,25 @@ public void testAfterUnbindConsumer() {
 		verify(this.channelProvisioner).afterUnbindConsumer(this.consumerDestination);
 	}
 
+	@Test
+	public void producerSyncPropertyFalseByDefault() {
```

Comment:
If it's possible, I think it would be better to set the property via `spring.cloud.stream.gcp.pubsub.default.producer.sync` rather than accessing the properties class directly. Mainly to verify that `spring.cloud.stream.gcp.pubsub.default.producer.sync` is indeed the correct format of the property name (as I am not sure what the correct form is myself).

Reply:
Yes, I've been getting increasingly uncomfortable with the amount of mocking in this test. I'll either move this into an integration test, or turn this test into a light integration test with only a mocked pubsub template.

Reply:
This is the best I could do without testing SCS implementation details. There is still one mocked transition from our properties into a SCS wrapper, but at least it tests the property mapping correctly.